language/stdlib/modules/libra_system_2.move

```diff
@@ -246,12 +329,58 @@ module LibraSystem2 {
         let new_validator_config = ValidatorConfig2::get_config(validator_info.addr);
         // check if information is the same
         let config_ref = &mut validator_info.config;
+
         let configs_changed = (config_ref != &new_validator_config);
         *config_ref = *&new_validator_config;
+
         configs_changed
     }
 
     fun is_validator_(addr: address, validators_vec_ref: &vector<ValidatorInfo>): bool {
         !Vector::is_empty(&get_validator_index_(validators_vec_ref, addr))
     }
+
+    fun emit_discovery_set_change() acquires DiscoverySet {
+        let discovery_set_ref = borrow_global_mut<DiscoverySet>(0xD15C0);
+        Event::emit_event<DiscoverySetChangeEvent>(
+            &mut discovery_set_ref.change_events,
+            DiscoverySetChangeEvent {
+                new_discovery_set: *&discovery_set_ref.discovery_set,
+            },
+        );
+    }
+
+    // Discovery methods
+
+    public fun get_discovery_address(d: &DiscoveryInfo): &address {
+        &d.addr
+    }
+
+    public fun get_discovery_config(d: &DiscoveryInfo): &ValidatorConfig2::DiscoveryConfig {
+        &d.discovery_config
+    }
+
+    // Get the DiscoveryInfo for the ith validator
+    public fun get_ith_discovery_info(i: u64): DiscoveryInfo acquires DiscoverySet {
+        let discovery_vec_ref = &borrow_global<DiscoverySet>(0xD15C0).discovery_set;
+        Transaction::assert(i < Vector::length(discovery_vec_ref), 4);
```

Comment:
still very curious about why we are asserting something the language will already do